timelock-impl/src/main/java/com/palantir/atlasdb/timelock/AsyncTimelockResource.java

```diff
@@ -125,9 +131,45 @@ public void waitForLocks(@Suspended final AsyncResponse response, WaitForLocksRe
     @POST
     @Path("refresh-locks")
     public Set<LockToken> refreshLockLeases(Set<LockToken> tokens) {
+        return timelock.refreshLockLeases(tokens).refreshedTokens();
+    }
+
+    @POST
+    @Path("lock-v2")
+    public void lockV2(@Suspended final AsyncResponse response, LockRequest request) {
+        AsyncResult<LeasableLockToken> result = timelock.lock(request);
+        lockLog.registerRequest(request, result);
+        result.onComplete(() -> {
+            if (result.isFailed()) {
+                response.resume(result.getError());
+            } else if (result.isTimedOut()) {
+                response.resume(LeasableLockResponse.timedOut());
+            } else {
+                response.resume(LeasableLockResponse.successful(result.get().token(), result.get().lease()));
+            }
+        });
+    }
+
+    @POST
+    @Path("start-atlasdb-transaction-v3")
+    public LeasableStartAtlasDbTransactionResponse startAtlasDbTransactionV3(
```

Comment:
probably worth grouping the logic in this class? so like, all the startAtlasDbTransactions end up near each other?

Also, I'd be tempted to call the old ones deprecatedStartAtlasDbTransaction? rather than versioning these method calls?